src/store/slices/uiSlice.js

```diff
@@ -0,0 +1,25 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+    isModalOpen: false
+}
+
+const uiSlice = createSlice({
+    name: 'ui',
+    initialState,
+    reducers: {
+        //TODO: we need to choose if we want to use toggle or open/close actions.
+        openModal(state) {
```

Comment:
u dont need the openmodal and closemodal actions if u have toggleModal wich will take care of modal open/closed state

Reply:
That's right, I only did this to offer two ways of handling that issue to @amirg76 , So he can decide which one is more suitable for the modal he created.